pkg/cfn/manager/create_tasks.go

```diff
@@ -80,6 +80,16 @@ func (c *StackCollection) NewUnmanagedNodeGroupTask(nodeGroups []*api.NodeGroup,
 	return taskTree
 }
 
+// NewKarpenterTask defines tasks required to create Karpenter
+func (c *StackCollection) NewTasksToInstallKarpenter() *tasks.TaskTree {
+	taskTree := &tasks.TaskTree{Parallel: true}
+	taskTree.Append(&karpenterTask{
+		info:            fmt.Sprintf("create karpenter for stack %q", c.spec.Metadata.Name),
+		stackCollection: c,
+	})
+	return taskTree
+}
+
```

Comment:
can the task live in the `actions/karpenter` package? We did something similar with addons https://github.com/weaveworks/eksctl/blob/main/pkg/actions/addon/tasks.go- I know the other tasks are in `manager`, but I don't really understand why

Reply:
Sure, I don't see why not.

Reply:
Yeah, this is actually really hard. Too many unexported things, and a circular dependency once you try removing any of the task creation out of manager.